cpp/log/cert_submission_handler.cc

```diff
@@ -64,6 +64,92 @@ CertSubmissionHandler::CertSubmissionHandler(const CertChecker* cert_checker)
     : cert_checker_(CHECK_NOTNULL(cert_checker)) {
 }
 
+// static
+StatusOr<size_t> CertSubmissionHandler::X509ChainToEntries(
```

Comment:
I think this is returning something along the lines of:
`[chain[0]]` when the leaf has no embedded certs, and
`[chain[0], ToPrecert(chain[0]), chain[1]]` otherwise (with the last element possibly overwritten if there's an intermediate higher up the chain with the precert signer NID).

It feels a bit weird to be returning both in the latter case, would it be a less confusing API if it were
`StatusOr<size_t> X509ChainToLogEntries(chain ..., LogEntry *x509_entries, vector<LogEntry>* precert_entries) `?

Reply:
done

Reply:
Btw. shouldn't I change:
"LogEntry *x509_entries"
to
"LogEntry *x509_entry"
?
This entry is always exactly one.

Reply:
Ah, yes - sorry!

Reply:
done